org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/util/JRTUtil.java

```diff
@@ -479,21 +450,19 @@ boolean sameRelease(String other) {
 		return Long.compare(jdkLevel, otherJdkLevel) == 0;
 	}
 
-	static String toJdkHome(File jrt) {
-		String home;
+	static Path toJdkHome(File jrt) {
 		Path normalized = jrt.toPath().normalize();
 		if (jrt.getName().equals(JRTUtil.JRT_FS_JAR)) {
-			home = normalized.getParent().getParent().toString();
+			return normalized.getParent().getParent();
 		} else {
-			home = normalized.toString();
+			return normalized;
 		}
-		return home;
 	}
 
-	static String readJdkReleaseFile(String javaHome) throws IOException {
+	static String readJdkReleaseFile(Path javaHome) throws IOException {
 		Properties properties = new Properties();
-		try(FileReader reader = new FileReader(new File(javaHome, "release"))){ //$NON-NLS-1$
-			properties.load(reader);
+		try (InputStream in = Files.newInputStream(javaHome.resolve("release"))) { //$NON-NLS-1$
```

Comment:
The `java.util.Properties` class states in its doc that it expect the content to be encoded using the `ISO 8859-1` encoding. Using a bare `InputStream` allows the implementation to handle it as it as expected instead of relying on a Reader that in this case uses the default encoding (which is UTF-8 in the latest JDKs).

That being said, I assume for the value read, the encoding should not really matter since it is simple ASCII.